spring-cloud-gateway-core/src/test/java/org/springframework/cloud/gateway/filter/GatewayMetricFilterTests.java

```diff
@@ -46,8 +42,8 @@
 import io.micrometer.core.instrument.MeterRegistry;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest(webEnvironment = RANDOM_PORT)
-@DirtiesContext
+@SpringBootTest(webEnvironment = DEFINED_PORT, properties = "server.port: ${test.fixed-port}")
```

Comment:
A fixed port will eventually fail. Do something like

```
	public int port;

	@Before
	public void init() {
		port = SocketUtils.findAvailableTcpPort();
		System.setProperty("server.port", String.valueOf(port));
	}
	
	@After
	public void after() {
		System.clearProperty("server.port");
	}
```

Reply:
Sorry for causing that many findings. Thanks for your feedback and the hint. This will unfortunately not work IMHO as I need to provide the port somehow already in the route definition URI for the comparison and setting the server.port at this stage is too late for that. I will look into it how I can somehow construct a proper test case with a random port.

Reply:
I do this all of the time. It's available before the application context is even constructed. I don't understand why this won't work. You can change server.port to test.fixed-port. It shouldn't matter where it is defined.

Reply:
I tried it quickly and the ${server.port} in the route definition URI in the test yml resolved to 0. So I thought that the context is created before. But then I must have done something wrong. I will give it another more accurate try tomorrow. Thx!

Reply:
It needs to stay `DEFINED_PORT`

Reply:
Thanks, works now with <code>DEFINED_PORT</code> and with setting the port before the web clients got initialized in the extended base class.